src/scala/org/pantsbuild/zinc/analysis/AnalysisMap.scala

```diff
@@ -65,13 +66,22 @@ class AnalysisMap private[AnalysisMap] (
      * Can remove after the sbt jar output patch lands.
      */
     def definesClass(classpathEntry: File): DefinesClass = {
-      getAnalysis(classpathEntry).map { analysis =>
-        // strongly hold the classNames, and transform them to ensure that they are unlinked from
-        // the remainder of the analysis
-        val classNames = analysis.asInstanceOf[Analysis].relations.srcProd.reverseMap.keys.toList.toSet.map(
-          (f: File) => filePathToClassName(f))
-        new ClassNamesDefinesClass(classNames)
-      }.getOrElse {
+      // If we have analysis with a valid Compilation, use the classnames it refers to.
+      val analysisDefinesClass =
+        for (
+          abstractAnalysis <- getAnalysis(classpathEntry);
+          analysis = abstractAnalysis.asInstanceOf[Analysis];
+          compilation <- analysis.compilations.allCompilations.headOption;
+          singleOutput <- compilation.getOutput.getSingleOutput.asScala;
```

Comment:
is this `getSingleOutput` always single? I think so because we're only asking for the analysis of one file, but still I want to make sure.

Reply:
Yea, this is effectively a dead API in zinc.